Create AutoUpdateLastCostExamples.vb

sybethanlevy opened the pull request.

sybethanlevy committed:
Create AutoUpdateLastCostExamples.vb

sybethanlevy commented on BreakpointSamples/Automation of Data Entry or Data Selection Processes/AutoUpdateLastCostExamples.vb:
```diff
@@ -0,0 +1,30 @@
+  Public Sub Invoke(ByVal transaction As Sybiz.Vision.Platform.Inventory.Transaction.ManufactureIssue, ByVal e As Sybiz.Vision.Platform.Admin.Breakpoints.BreakpointEventArgs) 'Do not remove - SYBIZ
+
+    For each l As Sybiz.Vision.Platform.Inventory.Transaction.ManufactureIssueLine In transaction.Lines
+      Sybiz.Vision.Platform.Inventory.Product.UpdateLastCost(l.Product,l.UnitCost,true)
```
Not sure if there's anything else that needs considering in relation to manufacturing? Couldn't think of anything though...

Marjon80 replied:
This is probably a good spot where you might want to not throw an exception, if doing a manufacture where a location is in negative stock this would explode and that might not be desired behavior; update last cost where possible and ignore it where it cannot.

sybethanlevy commented on BreakpointSamples/Automation of Data Entry or Data Selection Processes/AutoUpdateLastCostExamples.vb:
```diff
@@ -0,0 +1,30 @@
+
+    For each l As Sybiz.Vision.Platform.Inventory.Transaction.StockTransferLine In transaction.Lines
+      'If source location is elsewhere...
+      If l.SourceLocationDetails.Id = 0 Then
```
Since one can't go from elsewhere to job / service just kept it simple but could expand to consideration of X to job scenarios if wanted.

Marjon80 replied:
For each l As Sybiz.Vision.Platform.Inventory.Transaction.StockTransferLine In transaction.Lines
      'If source location is elsewhere...
      If  l.TransferType = Sybiz.Vision.Platform.Core.Enumerations.TransferType.LocationToLocation AndAlso l.Source = -1 Then		
		Try 
			Sybiz.Vision.Platform.Inventory.Product.UpdateLocationLastCost(l.Product,l.Destination,l.UnitCost,true)
		Catch ex as Exception
			BreakpointHelpers.ShowErrorMessage(e.Form, String.Format("Unable to update {0} last cost", l.ProductDetails.Code), ex.Message)
		End Try		
      End If
	  Next

Marjon80 replied:
This will let the system continue with other lines and is the expect use of throwing exceptions to advise the user that something couldn't work as expected.

sybethanlevy committed:
Update AutoUpdateLastCostExamples.vb

Adjusted considering comments.

Question: is checking for location to location necessary in the context that the only way "elsewhere" is invoked IS via Location to Location? (Also noting that Elsewhere is 0 not -1, which is the Virtual Job location.)

sybethanlevy commented:
Ref [9445bca](https://github.com/Sybiz/Vision/pull/4/commits/9445bca4fa23a14e95e5b33a1e17cdb5689f46e6) for further changes based on above comments.
